contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
I found [this](https://github.com/premake/premake-core/wiki/filter) entry to document that feature pretty well, should there be something more explicit, or obvious?

While I agree that a smaller list makes more sense to manage, those code blocks are off by default in Lua and we should follow that behaviour. Additionally, I'm going to assume that means the code that is compiled without those defines will work on all platforms, including Linux, BSD, Hurd and AIX. That's probably a terrible assumption to make though.